### ISSUE TITLE
RightCurly with "alone" option and "LITERAL_DO" token doesn't raise a violation for empty do block

### ISSUE THREAD

**Issue by guillaumecle** [checkstyle/checkstyle#3603]
javac Main.java

No output

cat Main.java

```java
public class Main{

	public static void main(String[] args){
		do{
		}while(System.currentTimeMillis() < 0);
	}

}
```
cat config.xml

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
		  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
		  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
	<property name="charset" value="UTF-8"/>

	<module name="TreeWalker">
		<module name="RightCurly">
			<property name="option" value="alone"/>
			<property name="tokens" value="LITERAL_DO"/>
		</module>
	</module>
</module>
```

java -jar checkstyle-7.3-all.jar -c config.xml Main.java 

```
Starting audit...
Audit done.
```

I expect the audit to raise a violation as the `while` is on the same line as the `}`. This is working correctly when the do block is not empty.

**Comment by rnveach** [comment]
> This is working correctly when the do block is not empty.

It looks like this is intentional, skipping of empty block. First introduced in Issue #213 and [seen in the code](https://github.com/checkstyle/checkstyle/blob/25a37e5049b7816b34c552899841a978efc37a63/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java#L253) for `ALONE` and `ALONE_OR_SINGLELINE`.

**Comment by MEZk** [comment]
@romani @rnveach 
This duplicates https://github.com/checkstyle/checkstyle/issues/3775 . 
Should be closed.

**Comment by romani** [comment]
code example is merged to #3775 .
